modules/react/popup/lib/PopupCard.tsx

```diff
@@ -71,9 +71,18 @@ export const PopupCard = createSubcomponent('div')({
     return getTransformFromPlacement(model.state.placement || 'bottom');
   }, [model.state.placement]);
 
+  const getMargin = (margin?: string | number) => {
+    if (margin) {
+      const result = space[margin as CanvasSpaceKeys];
+      // Get the top margin from "25px 50px" or "10px 15px 20px" format.
+      const marginTop = typeof margin === 'string' ? margin.split(' ')[0] : margin;
+      return result ? result : marginTop;
+    }
+    return space.xl;
+  };
+
```

Comment:
Retrieve the `top` margin when multiple values are provided. I don't know what is the best way to achieve this unless we create a new prop just for the `maxHeight` calculation and users would have to explicitly pass that in.
Also for this function, I assumed that we wanted the `top` value. Let me know if this needs to be changed otherwise.

Reply:
I appreciate you picking this up, @thunguyen19. 😄 I think this is generally the right idea, but I have some suggestions.

- Let's pull this function outside of the `PopupCard` – it doesn't need to live inside the component render cycle.
– Since this function is only used by `maxHeight`, let's pull all the calculations into one spot
– I think `marginTop` is a good rough proxy, but let's make this a little more robust to account for `marginBottom` was well.

Reply:
I wrote this somewhat quickly, so it might not work perfectly, but have a look at this:

```ts
function getSpace(value?: string | number) {
  if (value && value in space) {
    return space[value as keyof typeof space]
  } else {
    return value;
  }
};

// This does not account for explicit
function getMaxHeight(margin: PopupCardProps['margin']) {
  // set the default margin offset to space.xl
  let marginOffset: string | number = space.xl;

  if (margin) {
    // parse the margin prop
    if (typeof margin === 'string') {
      const marginValues = margin.split(' ');
      const marginTop = getSpace(marginValues[0]);
      // If provided, use the specific margin-bottom in the shorthand, otherwise use the margin-top value
      const marginBottom = getSpace(marginValues[2]) || marginTop;

      marginOffset = `calc(${marginTop} + ${marginBottom})`;
    } else {
      // if margin is a number, double it to get the offset
      marginOffset = margin * 2;
    }
  }
  // return the offset value
  return `calc(100vh - ${marginOffset})`;
}
```